### DISCUSSION TITLE
How to run a simple example like 'print kills'

### DISCUSSION THREAD

**Discussion by dr11m** [saul/demofile#324] [Questions]
I am new to node. I spent like several hours trying to run on windows a simple example like on a screenshot on a main page. All I need a simple example of how to make it or what kind of information should I search. Thanks!

**Comment by saul** [comment]
With a fresh Git checkout at `C:\Code\demofile`, do:

```
C:\Code\demofile> npm ci
C:\Code\demofile\examples> npm ci
C:\Code\demofile\examples> npx ts-node .\dumpfile.ts ..\demos\match730_003456990748161867873_0727052756_188.dem
```

Replace the path to the demo (e.g the `match730_003456990748161867873_0727052756_188.dem` part) with the path to a demo that exists on your machine.

Hope that helps.

**Reply by almondrubix** [reply]
Hi, I'm also new to node, and get a cannot find module error when I try .\npm ci

**Reply by saul** [reply]
Please can you post the complete output?

**Reply by almondrubix** [reply]
It's a different issue, but here is the output;

 `PS C:\> node node_modules\demofile\printKills.mjs
file:///C:/node_modules/demofile/printKills.mjs:1
import fs = require("fs");
          ^

SyntaxError: Unexpected token '='
    at Loader.moduleStrategy (internal/modules/esm/translators.js:145:18)
    at async link (internal/modules/esm/module_job.js:67:21)`

**Reply by BlackYuzia** [reply]
> import fs = require("fs"); 

Try:

```js
const fs = require("fs");
```

**Reply by almondrubix** [reply]
Hi Saul,
Following what you've written here, I run into the following issue:

```
PS C:\Users\Superadmin\demofile\examples> npx ts-node .\dumpfile.ts ..\demos\match730_003456990748161867873_0727052756_188.dem
npx: installed 14 in 13.719s
⨯ Unable to compile TypeScript:
dumpfile.ts:61:9 - error TS2532: Object is possibly 'undefined'.

61         terrorists.teamName,
           ~~~~~~~~~~
dumpfile.ts:62:9 - error TS2532: Object is possibly 'undefined'.

62         terrorists.clanName,
           ~~~~~~~~~~
dumpfile.ts:63:9 - error TS2532: Object is possibly 'undefined'.

63         terrorists.score,
           ~~~~~~~~~~
dumpfile.ts:64:9 - error TS2532: Object is possibly 'undefined'.

64         cts.teamName,
           ~~~
dumpfile.ts:65:9 - error TS2532: Object is possibly 'undefined'.

65         cts.clanName,
           ~~~
dumpfile.ts:66:9 - error TS2532: Object is possibly 'undefined'.

66         cts.score
           ~~~
dumpfile.ts:191:15 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

191 parseDemoFile(process.argv[2]);
```
